compactor/periodic.go

```diff
@@ -73,24 +77,20 @@ func (t *Periodic) Run() {
 					continue
 				}
 			}
-
-			if clock.Now().Sub(last) < executeCompactionInterval {
```

Comment:
While we're improving this function. Can move the remaining code in the for loop into the `case <-clock.After` for readability?  It's only ever run as part of that `case`...

Reply:
can we refactor the code in a future pr?